### DISCUSSION TITLE
Actividad 4 - GRUPO 14

### DISCUSSION THREAD

**Discussion by Lele3002** [frlp-utn-ingsoft/catedra#166] [Show and tell]
https://github.com/JuanbyGuada/vetsoft

**Comment by ivanduva** [comment]
Buenas! Les puedo pedir el link a donde lo deployaron?

**Reply by JuanbyGuada** [reply]
Sisi, sería este:

https://vetsoft-app-2-0-sad8.onrender.com/

**Reply by ivanduva** [reply]
Buenas,
Les vamos a pedir algunas correcciones, más que nada porque vemos muchos problemas de procedimiento:

- Los tests e2e fallan en main y hay diferencias entre develop y main. Debería quedar todo correcto en main. Tal vez no haya ayudado tener esa capa intermedia, revisen si es necesario tener develop o si les conviene sacarla.
- En cuanto a la validación del teléfono, les pedimos que agreguen un test e2e y otro de integración.
- En cuanto a la validación del nombre, les pedimos que agreguen un test e2e y otro de integración.
- Vemos que se agregó la regla d103 de ruff directamente en develop. Esto no está bien por varias razones que más adelante explicamos. Le pedimos a Aylen que agregue otra de las reglas que se encuentre en la consigna de la Actividad con el proceso correspondiente, a través de un PR.
- Luego de arreglar todo, saquen una release.

Recuerden que los cambios deben ser realizados por las personas que se encargaron de cada issue.

Después les dejamos algunos comentarios sobre procedimiento que deberían revisar en las próximas correcciones:

- En las reviews no hay comentarios con relevancia. Como dijimos más de una vez, comentarios del estilo "Esta completo, lo acepto" no aportan a la review. Para eso se pone el approve y listo.
- Vimos que hicieron commits sobre main o develop para agregar lo relativo a la CI/CD y otras cuestiones. Si estamos usando el workflow Feature Branch, esto es peligroso y constituye una muy mala práctica. Estos commits no pasan por una instancia formal de review y por lo tanto es poco seguro, puede romper la aplicación. Hay que hacer PRs para cumplir con el proceso.
- Vemos varios malos mensajes de commit, en algunos casos son poco explicativos y en otros se mencionan cuestiones relativas al código o a los archivos, lo cual es redundante porque esa info ya está en el commit. Recuerden que un mensaje de commit debería contener una síntesis de lo que ese commit soluciona en cuanto a funcionalidad.

Arreglando lo que les comentamos más arriba ya tendrían el trabajo integrador completamente aprobado.

**La última fecha límite de será el último día de cursada antes de las vacaciones: Lunes 8 de Julio 23:59 hs.**

Saludos

**Reply by Lele3002** [reply]
Buenas profe! Le comento que ya arreglamos los problemas que nos mencionaste. Saludos.

**Comment by ivanduva** [comment]
Perfecto! Queda aprobado el trabajo integrador.
Felicitaciones por el buen trabajo

Saludos
